tensorzero-core/tests/e2e/config.rs

```diff
@@ -973,3 +973,104 @@ model = "test_model_{random_id}"
         "Function from snapshot should still be present"
     );
 }
+
+/// Test that tags are merged when writing the same config snapshot twice.
+#[tokio::test(flavor = "multi_thread")]
+async fn test_write_config_snapshot_tag_merging() {
+    // Get a clean ClickHouse instance with automatic cleanup
```

Comment:
is this supposed to be clean? there's `get_clean_clickhouse`. It seems like you don't actually need a clean one (but in which case let's delete the migration logic since it's confusing)?